src/observer.js

```diff
@@ -299,18 +301,12 @@ export function observer(componentClass) {
         !componentClass.isReactClass &&
         !Component.isPrototypeOf(componentClass)
     ) {
-        const observerComponent = observer(
-            class extends Component {
-                static displayName = componentClass.displayName || componentClass.name
-                static contextTypes = componentClass.contextTypes
-                static propTypes = componentClass.propTypes
-                static defaultProps = componentClass.defaultProps
-                render() {
-                    return componentClass.call(this, this.props, this.context)
-                }
-            }
-        )
+        const observerComponent = observerLite(componentClass)
+        // TODO: move to mobx-react-lite
```

Comment:
I am fine with that. Regarding hoistStatics there is an open issue with some thoughts: https://github.com/mobxjs/mobx-react-lite/issues/32

Reply:
Done